### PR TITLE
Improve the support for @keyframes

### LINKED ISSUE #157: comments inside keyframe declarations highlight incorrectly
comments-inside-keyframes

### PR DESCRIPTION
**Pull request by esdoppio**
fixes #157

### CODE REVIEW

**Comment by esdoppio** [comment]
`@keyframes` is actually a CSS feature, probably I should change all the `*.scss` to `*.css`?

File: grammars/scss.cson [hunk 1]
```diff
@@ -329,48 +329,92 @@
...
+      'beginCaptures':
+        '0':
+          'name': 'keyword.control.at-rule.keyframes.scss'
+        '1':
+          'name': 'punctuation.definition.keyword.scss'
+      'end': '(?<=})'
+      'name': 'meta.at-rule.keyframes.scss'
+      'patterns': [
+        {
+          'match': '(?<=keyframes[ \\t])(?:[_A-Za-z][-\\w]|-[_A-Za-z])[-\\w]*'
```
Comment (50Wliu) [inline grammars/scss.cson]: I'm not liking this lookbehind because it doesn't take into account multiple spaces.

**Comment by 50Wliu** [comment]
My biggest issue with this is the use of lookbehinds.

File: grammars/scss.cson (hunk 1 above)
Comment (esdoppio) [inline grammars/scss.cson]: Yes, it can be, if you don't want to impose the "must contain at least 2 characters" restriction, coz it also matches something like `_`.

**Comment by esdoppio** [comment]
Yeah, I was being sloppy with that lookbehind, but I couldn't think of another way to do this. Do you have a better idea?

**Comment by esdoppio** [comment]
Better?

File: grammars/scss.cson (hunk 1 above)
Comment (50Wliu) [inline grammars/scss.cson]: `(?:-?[_A-Za-z])[-\\w]+` then?

**Comment by 50Wliu** [comment]
.scss can be kept for now.

**Comment by esdoppio** [comment]
Anything else?

**Comment by 50Wliu** [comment]
Why is `[[:blank:]]` being used over `\\s+`?

**Comment by esdoppio** [comment]
TL;DR. Performance. But I just did a benchmark, turns out `\s` is slightly faster, bad decision. :blush: Changing it back to `\s` for brevity.

My thoughts:

`\s` is a shorthand for `[\x20\t\v\f\n\r]`, `[[:blank:]]` is equivalent to `[\x20\t]`.

* `\v` and `\f` are basically never used in source code.
* `\n` and `\r` are used in source code, but due to the way how a grammar package works—*the regular expressions are matched against only a single line of the document at a time*—something like `[\n\r]@keyframes` can never match since `\n\r` are always at the end of the previous line.

So `[[:blank:]]` should be enough for tokenizing source code in Atom. I thought due to internal optimizations, `[[:blank:]]` and `\s` should be faster than `[\x20\t]`, and `[[:blank:]]` may be faster than `\s` (I was WRONG).

File: grammars/scss.cson (hunk 1 above)
Comment (esdoppio) [inline grammars/scss.cson]: 😆  That imposes a new restriction that if the identifier starts with `-`, it must contain at least 3 characters.

**Comment by 50Wliu** [comment]
Last thing: Replacing the space with `\x20` seems to obfuscate things.  Did you replace them because ` ` tends to get easily lost when scanning?

**Comment by esdoppio** [comment]
Still think `\x20` is better than ` `. :laughing: See that "void" over there?
